Task1/aga/IsPrime/IsPrime/Sieve.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IsPrime
+{
+    internal class Sieve
+    {
+        private int _max;
+
+        public Sieve(int val)
+        {
+           _max=val;
+        }
+        
+        private bool[] MakeSieve()
+        {
+            bool[] isPrime = new bool[_max + 1];
+            for (int i = 2; i<=_max; i++) { isPrime[i] = true; }
+            for (int i = 2; i<=_max;i++)
+            {
+                if (isPrime[i])
+                {
+                    for (int j = i * 2; j<=_max; j+=i)
+                    {
+                        isPrime[j] = false;
+                    }
+                }
+            }
+            return isPrime;
+        }
+
+        public List<int> GetPrimes()
+        {
+            bool[] primes = MakeSieve();
```

Comment:
In this way the sieve will be recreated every time GetPrimes() is called regardless it will be identicall each time (_max value won't change as there's no setter).
My preferred choice would be to make the _max_ a parameter of the GetPrimes() method (not a member).
Otherwise let's remember the list of primes after first creation.